### PR TITLE
Add thermal control daemon to monitor FAN and thermal status and run thermal policy

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
What I did?

1. Add FanUpdater to get fan information and write it to database
2. Add TemperUpdater to get temperature information and write it to database
3. Add ThermalControlDaemon to run the thermal policy

How to verify?

1. show platform fanstatus
2. show platform temperature
3. View syslog make sure thermalctld is up and there is no error logs

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@stepanblyschak please review as well

File: sonic-thermalctld/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[aliases]
+test=pytest
```
Comment (jleveque) [inline sonic-thermalctld/setup.cfg]: Add newline at EOF

File: sonic-thermalctld/scripts/thermalctld [hunk 2]
```diff
@@ -0,0 +1,570 @@
...
+    """
+    Handy function to invoke the callback and catch NotImplementedError
+    :param callback: Callback to be invoked
+    :param default: Default return value if exception occur
+    :return: Default return value if exception occur else return value of the callback
+    """
+    try:
+        ret = callback()
+    except NotImplementedError:
+        ret = None
```
Comment (jleveque) [inline sonic-thermalctld/scripts/thermalctld]: Why not just set `ret = default` here, then simply `return ret` below?

File: sonic-thermalctld/scripts/thermalctld [hunk 3]
```diff
@@ -0,0 +1,570 @@
...
+        status = temperature < threshold
+        if status == self.under_temperature:
+            return False
+
+        self.under_temperature = status
+        return True
+
+
+#
+# TemperUpdater ======================================================================
```
Comment (jleveque) [inline sonic-thermalctld/scripts/thermalctld]: `TemperatureUpdater` to match class name below

File: sonic-thermalctld/scripts/thermalctld [hunk 4]
```diff
@@ -0,0 +1,570 @@
...
+    :param callback: Callback to be invoked
+    :param default: Default return value if exception occur
+    :return: Default return value if exception occur else return value of the callback
+    """
+    try:
+        ret = callback()
+    except NotImplementedError:
+        ret = default
+
+    return default if ret is None else ret
```
Comment (jleveque) [inline sonic-thermalctld/scripts/thermalctld]: You made half of the change I requested. This line can now simply be changed to `return ret`

Comment (Junchao-Mellanox) [reply sonic-thermalctld/scripts/thermalctld]: Fixed

**Comment by jleveque** [comment]
@stepanblyschak to review as well, per Liat above.

**Comment by wendani** [comment]
Request for 201911 with Azure/sonic-buildimage#3949 @jleveque @abdosi

**Pull request merged**